examples/cpp-dummy/micro_cpp_dummy.cpp

```diff
@@ -69,6 +69,21 @@ void MicroSimulation::reload_checkpoint()
     _micro_scalar_data = _checkpoint;
 }
 
+// This function needs to set the complete state of a micro simulation
+void MicroSimulation::setState(double micro_scalar_data, double checkpoint)
```

Comment:
Do we here also need `sim_id` and `micro_vector_data?`

Reply:
This is just to show an example to show the user on how to use the functions. We don't actually need to set and get the full state of a dummy problem. Hence I have purposely kept things simple here.